src/manager/file.vala

```diff
@@ -0,0 +1,71 @@
+namespace ProtonPlus.Manager {
+    public class File {
+        public static void Extract (string install_location, string launcher_name, string tool_name) {
+            Archive.Read archive = new Archive.Read ();
+            archive.support_format_all ();
+            archive.support_filter_all ();
+
+            int flags;
+            flags = Archive.ExtractFlags.ACL;
+            flags |= Archive.ExtractFlags.PERM;
+            flags |= Archive.ExtractFlags.TIME;
+            flags |= Archive.ExtractFlags.FFLAGS;
+
+            Archive.WriteDisk ext = new Archive.WriteDisk ();
+            ext.set_standard_lookup ();
+            ext.set_options (flags);
+
+            if (archive.open_filename (install_location + tool_name + ".tar.gz", 1920000) != Archive.Result.OK) return;
+
+            ssize_t r;
+
+            unowned Archive.Entry entry;
+
+            int bob = 0;
+            string test = "";
+
+            for ( ;; ) {
+                r = archive.next_header (out entry);
+                if (r == Archive.Result.EOF) break;
+                if (r < Archive.Result.WARN) break;
+                entry.set_pathname (install_location + entry.pathname ());
+                if(bob++ == 0){
```

Comment:
The condition is only true on the first iteration. A boolean flag would be clearer.

Reply:
C'etait pour tester, meme chose pour les variables bob et test.